cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -408,11 +408,31 @@ def test_suspend(self, verify_vm_running, vm):
         vm.action.suspend()
         wait_for(lambda: vm.power_state == "suspended", num_sec=300, delay=5, fail_func=vm.reload)
 
+    def test_reset(self, verify_vm_running, vm):
+        assert "reset" in vm.action
+        vm.action.reset()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
+
+    def test_restart(self, verify_vm_running, vm):
+        assert "restart" in vm.action
+        vm.action.restart()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
+
+    def test_shutdown(self, verify_vm_running, vm):
+        assert "shutdown" in vm.action
+        vm.action.shutdown()
+        wait_for(lambda: vm.power_state == "off", num_sec=300, delay=5, fail_func=vm.reload)
+
+    def test_retire(self, verify_vm_running, vm):
+        assert "retire" in vm.action
+        vm.action.retire()
+        wait_for(lambda: vm.power_state == "off", num_sec=300, delay=5, fail_func=vm.reload)
+
 
 @pytest.mark.usefixtures("test_vm")
 @pytest.mark.usefixtures("setup_provider_clsscope")
 @pytest.mark.uncollectif(lambda: appliance_is_downstream() and current_version() < "5.4")
-class TestDeleteViaREST(object):
+class TestGrudViaREST(object):
```

Comment:
Not a big deal but I think a typo happened here:`TestGrudViaREST` => `TestCrudViaREST` :wink: